### PR TITLE
Horizontal Pod Autoscaler

### PR DESCRIPTION
**Pull request by rowanruseler**
Signed-off-by: Rowan Ruseler <rowanruseler@gmail.com>

### CODE REVIEW

File: charts/pgadmin4/templates/hpa.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  minReplicas: {{ .Values.autoscaling.minReplicas }}
+  maxReplicas: {{ .Values.autoscaling.maxReplicas }}
+  metrics:
+    {{- if .Values.autoscaling.targetCPUUtilizationPercentage }}
+    - type: Resource
+      resource:
+        name: cpu
+        targetAverageUtilization: {{ .Values.autoscaling.targetCPUUtilizationPercentage }}
+    {{- end }}
+    {{- if .Values.autoscaling.targetMemoryUtilizationPercentage }}
```
Comment (vutny) [inline charts/pgadmin4/templates/hpa.yaml:22]: This option is not covered in the README.

**Review by vutny — CHANGES_REQUESTED** [review]

File: charts/pgadmin4/templates/hpa.yaml (hunk 1 above)
Comment (rowanruseler) [reply charts/pgadmin4/templates/hpa.yaml:22]: Ah nice catch, thanks!

File: charts/pgadmin4/README.md [hunk 2]
```diff
@@ -68,27 +68,31 @@ The command removes nearly all the Kubernetes components associated with the cha
...
-| `resources` | CPU/memory resource requests/limits | `{}` |
 | `livenessProbe` | [liveness probe](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/) initial delay and timeout | `` |
 | `readinessProbe` | [readiness probe](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/) initial delay and timeout | `` |
 | `VolumePermissions.enabled` | Enables init container that changes volume permissions in the data directory  | `false` |
 | `extraInitContainers` | Init containers to launch alongside the app | `[]` |
+| `resources` | CPU/memory resource requests/limits | `{}` |
+| `autoscaling.enabled` | Enables Autoscaling | `false` |
+| `autoscaling.minReplicas` | Minimum amount of Replicas | `1` |
+| `autoscaling.maxReplicas` | Maximum amount of Replicas| `100` |
+| `autoscaling.targetCPUUtilizationPercentage` | Target CPU Utilization in percentage | `80` |
```
Comment (rowanruseler) [inline charts/pgadmin4/README.md:91]: @vutny it was already documented but I forgot to add the `autoscaling.` infront of it.

File: charts/pgadmin4/values.yaml [hunk 3]
```diff
@@ -251,6 +243,27 @@ extraInitContainers: |
...
+  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
+  # limits:
+  #   cpu: 100m
+  #   memory: 128Mi
+  # requests:
+  #   cpu: 100m
+  #   memory: 128Mi
+
+## Horizontal Pod Autoscaling
+## ref: https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/
```
Comment (rowanruseler) [inline charts/pgadmin4/values.yaml:259]: added reference point

**Pull request merged**